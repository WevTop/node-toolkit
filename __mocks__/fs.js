const fs = jest.genMockFromModule('fs')
const realFs = jest.requireActual('fs')

const fileHolder = new Map()

// Allow reading from disk.
fs.readdir = realFs.readdir
fs.readFile = realFs.readFile

// Write file contents to memory.
fs.writeFile = (path, contents, cb) => {
  fileHolder.set(path, contents)
  // In mock world, we can never fail. :')
  cb(null)
}

// Add some helper methods for getting and setting memory FS.
fs.__reset = () => fileHolder.clear()
fs.__getContents = path => fileHolder.get(path)

module.exports = fs
