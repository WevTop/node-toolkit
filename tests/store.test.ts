import fs from 'fs'
import path from 'path'
import Store from '../src/Store'

describe('Store', () => {
  const file = path.join(__dirname, 'tmp', 'map.txt')

  describe('#load', () => {
    it('loads a new map from a file', async () => {
      const store = new Store(path.join(__dirname, 'fixtures', 'map.txt'))
      expect(store).toBeInstanceOf(Store)
      expect(store).toMatchSnapshot()
    })

    it('does not load a new map from a file if the file does not exist', async () => {
      const store = new Store(path.join(__dirname, 'fixtures', 'nope.txt'))
      expect(store).toBeInstanceOf(Store)
      expect(store).toMatchSnapshot()
    })
  })

  describe('#write', () => {
    it('writes the entries to a file', async () => {
      const store = new Store()
      store.set('foo', true).set('bar', false)
      await store.write(file)
      expect(fs.existsSync(file)).toBe(true)
      expect(fs.readFileSync(file, 'utf8')).toMatchSnapshot()
    })

    afterEach(() => {
      if (fs.existsSync(file)) {
        fs.unlinkSync(file)
      }
    })
  })
})
