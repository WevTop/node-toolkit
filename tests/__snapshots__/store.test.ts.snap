// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store #constructor does not load a new map from a file if the file does not exist 1`] = `Map {}`;

exports[`Store #constructor loads a new map from a file 1`] = `
Map {
  "foo" => true,
  "bar" => false,
}
`;

exports[`Store #write writes the entries to a file 1`] = `"[[\\"foo\\",true],[\\"bar\\",false]]"`;
